libavfilter/vf_libvmaf.c

```diff
@@ -408,6 +549,81 @@ static int parse_models(AVFilterContext *ctx)
     return err;
 }
 
+static int parse_metadata_handlers(AVFilterContext *ctx)
+{
+    LIBVMAFContext *s = ctx->priv;
+    AVDictionary **dict;
+    unsigned dict_cnt;
+    int err = 0;
+
+    if (!s->metadata_feature_cfg)
+        return 0;
+
+    dict_cnt = 0;
+    dict = delimited_dict_parse(s->metadata_feature_cfg, &dict_cnt);
+    if (!dict) {
+        av_log(ctx, AV_LOG_ERROR,
+               "could not parse metadata feature config: %s\n",
+               s->metadata_feature_cfg);
+        return AVERROR(EINVAL);
+    }
+
+    for (unsigned i = 0; i < dict_cnt; i++) {
+        VmafMetadataConfiguration *metadata_cfg = av_calloc(1, sizeof(*metadata_cfg));
+        const AVDictionaryEntry *e = NULL;
+        char *feature_name = NULL;
+
+        while (e = av_dict_iterate(dict[i], e)) {
+            if (!strcmp(e->key, "name")) {
+                metadata_cfg->feature_name = av_strdup(e->value);
+                continue;
+            }
+        }
+
+        metadata_cfg->data = s->cb;
+        metadata_cfg->callback = &set_meta;
+
+        err = vmaf_register_metadata_handler(s->vmaf, *metadata_cfg);
```

Comment:
Anywhere we're using these new APIs, we'll want to detect if they are available and guard with macros. See CONFIG_LIBVMAF_CUDA_FILTER for an example. This macro is generated during the configure stage.

Reply:
The reason for this is compatibility with older libvmaf versions. fyi, these new apis are not part of a libvmaf release yet.

Reply:
https://github.com/yigithanyigit/FFmpeg/pull/2/commits/e037876711a6b105343239ecfe881e3ce3697f1c

Reply:
> Anywhere we're using these new libvmaf metadata apis, we'll want to detect if they are available and guard with macros. See CONFIG_LIBVMAF_CUDA_FILTER for an example. This macro is generated during the configure stage.

Should those changes need to own functions like `libvmaf_cuda` or putting guard macros like I did is fine?